src/main/java/net/avalara/avatax/rest/client/RestCall.java

```diff
@@ -157,55 +183,55 @@ public T call() throws Exception {
         long startTime = System.currentTimeMillis();
         T obj = null;
         String json = null;
-        CloseableHttpResponse response = null;
-        try {
-            response = this.client.execute(this.request);
-            HttpEntity entity = response.getEntity();
-            if (entity!=null)
-                json = EntityUtils.toString(entity);
-
-            if (response.getStatusLine().getStatusCode() / 100 != 2)
-            {
-                // populate error log info here
-                logObject.populateErrorInfo(json, response, startTime);
-                throw new AvaTaxClientException((ErrorResult) JsonSerializer.DeserializeObject(json, ErrorResult.class), model);
-            }
-            if (json != null) {
-                if (ContentType.getOrDefault(entity).getMimeType().equals("application/json") && !Objects.equals(typeToken.getType(), String.class)) {
-                    obj = (T) JsonSerializer.DeserializeObject(json, typeToken.getType());
-                } else {
-                    obj = (T) json;
+
+        try(CloseableHttpResponse response = client.execute(this.request)) {
```

Comment:
this use to do a try() on the ClosableHttpClient itself. This was causing an issue because it disposed of the client after every single call instead of letting the caller close the client when they are done making calls. That approach led to the thread pool being closed on subsequent calls.